evm/src/main/java/org/hyperledger/besu/evm/precompile/BLAKE2BFPrecompileContract.java

```diff
@@ -77,6 +83,31 @@ public PrecompileContractResult computePrecompile(
       return PrecompileContractResult.halt(
           null, Optional.of(ExceptionalHaltReason.PRECOMPILE_ERROR));
     }
-    return PrecompileContractResult.success(Hash.blake2bf(input));
+    PrecompileInputResultTuple res = null;
+    Integer cacheKey = null;
+    if (enableResultCaching) {
+      cacheKey = Arrays.hashCode(input.toArrayUnsafe());
+      res = blakeCache.getIfPresent(cacheKey);
+      if (res != null) {
+        if (res.cachedInput().equals(input)) {
+          cacheEventConsumer.accept(new CacheEvent(PRECOMPILE_NAME, CacheMetric.HIT));
+          return res.cachedResult();
+        } else {
+          cacheEventConsumer.accept(new CacheEvent(PRECOMPILE_NAME, CacheMetric.FALSE_POSITIVE));
+        }
+      } else {
+        cacheEventConsumer.accept(new CacheEvent(PRECOMPILE_NAME, CacheMetric.MISS));
+      }
+    }
+
+    res =
+        new PrecompileInputResultTuple(
+            input, PrecompileContractResult.success(Hash.blake2bf(input)));
+
+    if (cacheKey != null) {
+      blakeCache.put(input.hashCode(), res);
```

Comment:
I suggest using cacheKey as done in other precompiles instead of calculating the hashcode  again, and it is not done the same way.

Reply:
yeah this might be a problem if the input is getting mutated.  I missed this